src/main/java/chess/domain/piece/Piece.java

```diff
@@ -0,0 +1,38 @@
+package chess.domain.piece;
+
+import chess.domain.square.Direction;
+import chess.domain.square.Square;
+
+abstract public class Piece {
+
+    protected Team team;
+
+    protected Piece(final Team team) {
+        this.team = team;
+    }
+
+    public abstract Direction findDirection(final Square current, final Square destination);
```

Comment:
구현 해놓고 보니 모든 객체가 갭을 구하고 있네요 ㅎ
이런 경우 공통 코드는 부모 객체에서 해줘도 되겠죠
```suggestion
    public Direction findDirection(final Square current, final Square destination) {
        int fileDifference = current.getFileDifference(destination);
        int rankDifference = current.getRankDifference(destination);
        return judgePieceDirection(fileDifference, rankDifference);
    }

    protected abstract Direction judgePieceDirection(final int fileDifference, final int rankDifference)
```
이렇게 공통 구현부는 구현해두고 다른 부분만 구현하도록 하는 패턴을 템플릿 메서드 패턴 이라고 합니다 ㅎ
처음부터 공통부를 뽑으려고 하지말고 구현 하다가 공통부가 생기면 리팩토링 하는 방향이 좋아요

Reply:
반영했습니다 🙂
중복이 생긴다는 걸 느꼈었는데 외부에서 갭을 구하고 주입하는 것보단 낫다고 생각해서 리팩터링 하지 못했었습니다.
그런데 이렇게 메서드에서 추상메서드를 호출하는 방식으로 해결할 수 있었군요! 감사합니다 ㅎㅎ